smtpmail/backend/plugin/src/main/kotlin/com/ritense/valtimo/backend/plugin/client/SmtpMailClient.kt

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2015-2022 Ritense BV, the Netherlands.
+ *
+ * Licensed under EUPL, Version 1.2 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ritense.valtimo.backend.plugin.client
+
+import com.ritense.plugin.service.PluginService
+import com.ritense.resource.service.TemporaryResourceStorageService
+import com.ritense.valtimo.backend.plugin.dto.SmtpMailContentDto
+import com.ritense.valtimo.backend.plugin.dto.SmtpMailContextDto
+import com.ritense.valtimo.backend.plugin.dto.SmtpMailPluginPropertyDto
+import com.ritense.valtimo.backend.plugin.plugin.SmtpMailPlugin
+import mu.KotlinLogging
+import org.springframework.mail.javamail.JavaMailSender
+import org.springframework.mail.javamail.JavaMailSenderImpl
+import org.springframework.mail.javamail.MimeMessageHelper
+import javax.mail.internet.MimeMessage
+
+class SmtpMailClient(
+    private val pluginService: PluginService,
+    private val storageService: TemporaryResourceStorageService
+) {
+
+    fun sendEmail(
+        mailContext: SmtpMailContextDto,
+        mailContent: SmtpMailContentDto
+    ) {
+        try {
+            val javaMailSender = javaMailSender()
+
+            val message: MimeMessage = javaMailSender.createMimeMessage()
+
+            with(MimeMessageHelper(message, true)) {
+                setFrom(mailContext.sender.address)
+                mailContext.recipients.forEach { addTo(it.address) }
+                mailContext.ccList.forEach { addCc(it.address) }
+                mailContext.bccList.forEach { addBcc(it.address) }
+                setSubject(mailContext.subject)
+                setText(mailContent.mailMessage, true)
+                mailContent.attachments.forEach {
+                    addAttachment(it.fileName) { storageService.getResourceContentAsInputStream(it.fileResourceId) }
+                }
+
+            javaMailSender.send(message)
+            }
+        } catch (e: Exception) {
+            logger.warn {
+                "Sending email with subject ${mailContext.subject} has failed with error message ${e.message}"
+            }
```

Comment:
If its like this the error will be ignored and the service-task that should send the email will succeed even thought no mail was send.

Reply:
We (with Thomas and Frits) discussed that error handling is out of scope, since it is not an option to catch this in the process. So the alternative is crashing the process. Is that preferable? Or should we let this up to the implementation?

Reply:
Crashing the process is preferable. Because there must be a way to guarantee that an email has been send. It's then Valtimo's job to make it possible to catch the error with a 'Error Boundary Event' which doesn't work right now.